src/main/kotlin/no/universitetsforlaget/juridika/libraries/jsonml/JsonML.kt

```diff
@@ -31,13 +29,18 @@ sealed class JsonML {
             val tagName = jsonArray[0] as String
             var childStartIndex = 1
 
-            val attributes: Map<String, String>? = if (jsonArray.size > 1 && jsonArray[1] is JSONObject) {
+            val attributes: Map<String, Any>? = if (jsonArray.size > 1 && jsonArray[1] is Map<*, *>) {
```

Comment:
Gir det mening at value-typen i `attributes` er `Any`? Attributtverdier er vel alltid strenger?

Reply:
Det gir jo mening at det er any:
1. Det blir ikke String hvis vi encoder Hast-modellen i JsonML (attributtverdier kan i det tilfellet være Array)
2. Java har vel uansett ingen måte å statisk sjekke at noe som blir deserialisert til en generisk struktur som Map faktisk har de riktige typene. For generics i java er visst bare noe som finnes i compileren og blir informasjonen blir slettet når du er i bytecode?

Reply:
JsonML blir altså grunnleggende type-usikkert når det implementeres i Java/Kotlin, fordi du uansett må støtte `String | Array | Object` på ett eller annet nivå.

Reply:
Uansett, så er typen `Map<String, Any>` ca. ekvivalent med typen `JSONObject` (json-objekter kan ha enhver type som verdi i mapet).

Problemet her var at `JSONObject` er en faktisk _implementasjon_ av interfacet `Map<K, V>`. Og vi kan ikke anta at det er denne implementasjonen som blir opprettet, for det har vi jo overhodet ikke kontroll på selv.

Om ting blir deserialisert fra REST-JSON, er det vel Jackson som bestemmer hvilken Map/List-implementasjon som brukes. Om ting blir deserialisert fra `jdbc`, så er det sikkert noe annen kode.. Jeg vet ikke altså, men problemet var at det faktisk var `ArrayList` som var implementasjonen, mens den gamle koden forventet at `JSONArray` skulle være implementasjonen.